### PR TITLE
[EOC-61] Fix server issues

### PR DESCRIPTION
**Pull request by adamklepacz**
I did a bit of investigation before doing this task. I have found a nice article about error messages 
https://uxdworld.com/2018/05/30/how-to-write-good-error-messages/
Referring to the article all `err.message `sent from backend are replaced with custom messages, 
to make notifications understandable to the user. 

Previously using `err.message`,  we had something like this: 
Zrzut ekranu 2019-03-08 o 11 55 49
which was not understandable to a non-technical user. 
Now all the messages are easy to understand.

### CODE REVIEW

File: server/controllers/shoppingList.js [hunk 1]
```diff
@@ -236,10 +189,9 @@ const updateShoppingListItem = (req, resp) => {
    * are passed in the request
    *  */
   const propertiesToUpdate = {};
-  typeof isOrdered !== 'undefined'
-    ? (propertiesToUpdate['products.$.isOrdered'] = isOrdered)
-    : null;
-  voterIds ? (propertiesToUpdate['products.$.voterIds'] = voterIds) : null;
+  typeof isOrdered !== 'undefined' &&
+    (propertiesToUpdate['products.$.isOrdered'] = isOrdered);
+  voterIds && (propertiesToUpdate['products.$.voterIds'] = voterIds);
```
Comment (marekrozmus) [inline server/controllers/shoppingList.js]: Please do not use `&&` in this way - use simple `ifs`.
That kind of `&&` operator usage is ok for jsx.

File: server/controllers/shoppingList.js [hunk 2]
```diff
@@ -205,24 +158,24 @@ const getListData = (req, resp) => {
...
       }
 
-      if (documents && documents.length > 0) {
-        const { cohortId, _id, isArchived } = documents[0];
-
-        if (isArchived) {
-          return resp.status(200).json({ cohortId, _id, isArchived });
-        }
+      if (doc) {
+        const list = _head(doc);
```
Comment (alemikolo) [inline server/controllers/shoppingList.js]: We look exactly for one list, so we can use findOne and use doc without lodash _head an or checking docs.length

File: server/controllers/shoppingList.js [hunk 3]
```diff
@@ -205,24 +158,24 @@ const getListData = (req, resp) => {
...
-        return resp.status(200).json(documents[0]);
+        return isArchived
+          ? resp.status(200).json({ cohortId, _id, isArchived })
+          : resp.status(200).json(list);
       }
 
-      return resp
-        .status(404)
-        .send({ message: 'Products not found for given list id!' });
+      resp.status(404).send({ message: 'List data not found.' });
```
Comment (alemikolo) [inline server/controllers/shoppingList.js]: I think that we should check if doc exist first and use quick return if not. There will be more logic when we add checking roles, so all code for creating success response should be at the end...

File: server/controllers/shoppingList.js [hunk 4]
```diff
@@ -28,111 +32,51 @@ const createList = (req, resp) => {
...
-            message:
-              "Oops we're sorry, an error occurred while deleting the list"
+      doc
+        ? resp.status(200).send({
+            message: `List "${doc.name}" successfully deleted.`
           })
-        : resp.status(200).send({
-            message: `List ${doc.name} was successfully deleted!`
-          });
+        : resp.status(404).send({ message: `List "${doc.name}" not found` });
```
Comment (alemikolo) [inline server/controllers/shoppingList.js]: If !doc we can't access doc.name

File: server/controllers/shoppingList.js [hunk 5]
```diff
@@ -205,24 +158,22 @@ const getListData = (req, resp) => {
         { purchaserIds: req.user._id }
       ]
     },
-    (err, documents) => {
+    (err, doc) => {
       if (err) {
-        return resp.status(404).send({ message: err.message });
+        return resp.status(400).send({
+          message:
+            'An error occurred while fetching the lists data. Please try again.'
```
Comment (alemikolo) [inline server/controllers/shoppingList.js]: list data not lists - this method gets details of one list

**Pull request merged**